pkg/virt-controller/watch/vmi.go

```diff
@@ -1613,19 +1620,30 @@ func (c *VMIController) volumeReadyToAttachToNode(namespace string, volume virtv
 	} else if volume.PersistentVolumeClaim != nil {
 		name = volume.PersistentVolumeClaim.ClaimName
 	}
+
+	dataVolumeFunc := dataVolumeByNameFunc(c.dataVolumeInformer, dataVolumes)
```

Comment:
> There is one other check if the volume is populated for hotplug. Is that covered too?

I think it is, since `volumeReadyToAttachToNode` is invoked for every datavolume, be it hotplugged or not.

> It may make sense to get rid of this callback function and now, that we have to do pre-validation to see if the DV exist at all, and just pass in the data volume to IsPopulated. Since that is provided by CDI, it could probably be wrapped so that one can just pass in the DV. Not a big thing but the flow is pretty bumpy.

I'm not sure I entirely follow. Did you mean to change the code such that the dataVolumeFunc that is passed into `IsPopulated` will directly return the correct `DataVolume` object? If so, than this is not too far from what's already being done.